wallet/src/actors/worker/methods.rs

```diff
@@ -1009,16 +1015,12 @@ impl Worker {
 
     pub fn index_block(
         &self,
-        block: types::ChainBlock,
+        block: Arc<types::ChainBlock>,
```

Comment:
If a function does not use `Arc::clone`, prefer using a normal reference.

```suggestion
        block: &types::ChainBlock,
```

Reply:
What about now? :grimacing:

Reply:
Confused. I think we can remove the block clone if we change `block_txns` to be an iterator, and make `index_txns` accept an `impl Iterator<Item = &Transaction>`.

Reply:
Done!

Reply:
Now `index_block` does not need an `Arc<types::ChainBlock>`, only a reference

Reply:
But man, all of this code lives in the worker. The way for blocks to arrive to the worker is wrapped in an `Arc` so as to avoid cloning the block for each wallet that may be unlocked at that time (1 wallet = 1 worker). And once you put something into an `Arc`, turning it into a plain reference is probably not the best idea, right?

Reply:
Turning an arc into a reference is free. If this was a public function, then it would be more sense to use references because in that case we avoid creating arcs when we only have references. But if this is an internal function and we plan on using `Arc<Block>` everywhere, then there is no problem.